waspc/src/Wasp/Util/ShellCommand.hs

```diff
@@ -0,0 +1,22 @@
+module Wasp.Util.ShellCommand where
```

Comment:
This is quite cool, I never used `IsString`!

I like the concept you used here, it is quite neat. So basically we have this notion of Shell command argument, and we can say if it is quoted or not, and we carry this knowledge around, and then can act on it.

What I found tricky is:
1. There are no comments in this file explaining the reasoning behind it and what it is for. It is somewhat obvious, but I don't think it is obvious enough.
2. Continuing on above, I think the reason it is not obvious enough is that usecase feels quite trivial. If I want args quoted, why don't I continue using String and just add quotes to it? Ok, I guess this gives more knowledge, so you can avoid quoting already quoted arg, or you can unquote already quoted arg (although that feels a bit weird, why was it quoted then, but ok, maybe we know somebody else will quote it). Still, I think it is a bit confusing so explanation would help.
3. Using it above in the Jobs.hs feels a bit like overkill. Basically we need to quote these two args that are paths. And in one situation we need to quote them, in other we don't. But now we have to introduce this new type in all of the args there. But, not all of them, just some of them -> its not like we are using it across the whole codebase, just here. Feels weird that we are using it just in the small part of the codebase, how do we choose? When should I use this? Btw, isn't there a simpler solution -> can't we just quote those paths and that is it? Is it a problem if they are also quoted in that situation where they shouldn't be quoted, or will it just work?

Reply:
So, the problem we have is the following: we need the same command in a slightly different ways for Linux and for MacOS.

### MacOS

- The arguments should be a list of arguments
- The paths passed into the prisma command **MUST NOT** be quoted

### Linux

- The arguments are concatenated into a single string
- The paths **MUST** be quoted (since they can have spaces in them)

I'll try to simplify the solution and make it very local (it will only affect the array of prisma db migrate args and nothing else).

Reply:
Nice, this is much simpler now!

I left some small comments but otherwise it is good.

I would possibly consider skipping this file here, ShellCommand.hs, and just inlining this function `quoteArg` where it is used, since it is so niche / local, I don't think we benefit much from having it here in a separate file.